src/components/CommentSummary/index.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+
+import { gettext } from '../../utils';
+import { Comment, createCommentKey } from '../../reducers/comments';
+import styles from './styles.module.scss';
+
+export type PublicProps = { className?: string; comments: Comment[] };
+
+const CommentSummary = ({ comments, className }: PublicProps) => {
+  const overallComments = [];
+  const fileComments = [];
+
+  for (const comment of comments) {
+    if (comment.filename) {
+      fileComments.push(comment);
+    } else {
+      overallComments.push(comment);
+    }
+  }
+
+  const lines: string[] = [];
+
+  const addComment = (comment: Comment) => {
+    lines.push(comment.content || '');
+    lines.push('');
+  };
+
+  if (overallComments.length) {
+    lines.push(gettext('Overall comment(s):'));
+    lines.push('');
+
+    for (const comment of overallComments) {
+      addComment(comment);
+    }
+  }
+
+  const groups: { [key: string]: Comment[] } = {};
+
+  // Group comments by file / line so they appear under the same header.
+  for (const comment of fileComments) {
+    const key = createCommentKey({
+      fileName: comment.filename,
+      line: comment.lineno,
+    });
+
+    if (!groups[key]) {
+      groups[key] = [];
+    }
+
+    groups[key].push(comment);
+  }
+
+  for (const key of Object.keys(groups)) {
+    let addedHeader = false;
+
+    for (const comment of groups[key]) {
+      if (!addedHeader) {
```

Comment:
This will repeat the "File:" heading for every line in a given file, which seems like maybe not what we'd want. It seems like we only need the "File:" heading for the first comment in a given file, and only really need to repeat the line which has `filename#lineno` for each line's comment(s).

Reply:
Jorge's [notes](https://docs.google.com/document/d/1y4crrHxhGejb3eZ6HpjtBdDqONJakfwtfjaOqwq-qZo/edit#heading=h.28bv6iqbbwof) kind of sketched it out this way. I think it might be a little confusing without this header since filenames are not always obvious. In any case, I think this is a good starting point. We can get feedback from reviewers after they use it.